### PR TITLE
Image sitemap support

### PR DESCRIPTION
**Pull request by stefandoorn**
Rebased against master from #51.

- [ ] Add provider
- [ ] Add XML tests

### CODE REVIEW

**Comment by stefandoorn** [comment]
@Roshyo Moved your commits / PR here to continue on it. Also opened #64 to keep track of the additional todo's. Let me know your feedback. 

I've decided for now that the images can go in the same sitemap as the URL's instead of in it's own file, seems ok: https://support.google.com/webmasters/answer/178636?hl=en.

File: src/Model/SitemapUrlInterface.php [hunk 1]
```diff
@@ -35,10 +27,19 @@ public function getChangeFrequency(): string;
...
+     * @return Collection|SitemapImageUrlInterface[]
+     */
+    public function getImages(): Collection;
+
+    /**
+     * @param Collection|SitemapImageUrlInterface[] $images
+     */
+    public function setImages(Collection $images): void;
+
+    public function addImage(SitemapImageUrlInterface $image): void;
```
Comment (Roshyo) [inline src/Model/SitemapUrlInterface.php]: Wouldn't it be nice to also add a "removeImage" ?

File: src/Model/SitemapUrl.php [hunk 2]
```diff
@@ -116,4 +126,25 @@ public function setPriority(float $priority): void
...
+     */
+    public function setImages(Collection $images): void
+    {
+        $this->images = $images;
+    }
+
+    public function addImage(SitemapImageUrlInterface $image): void
+    {
+        $this->images->add($image);
+    }
```
Comment (Roshyo) [inline src/Model/SitemapUrl.php]: Wouldn't it be nice to also add a "removeImage" ?

File: src/Resources/views/Macro/xml.html.twig [hunk 3]
```diff
@@ -17,3 +17,25 @@
         <priority>{{ url.priority }}</priority>
     {% endif %}
 {%- endmacro %}
+
+{%- macro images(url) -%}
+    {%- if url.getImages is not empty -%}
+        {%- for image in url.getImages -%}
+            <image:image>
+                <image:loc>{{ image.location }}</image:loc>
+                {% if image.title is not empty %}
```
Comment (Roshyo) [inline src/Resources/views/Macro/xml.html.twig]: I'd also put all tags with "-" {%- if image.title is not empty -%}

File: src/Resources/views/show.xml.twig [hunk 4]
```diff
@@ -15,6 +15,7 @@
             {{ xml_helper.last_modification(url) }}
```
Comment (Roshyo) [inline src/Resources/views/show.xml.twig]: Same for those : {{- xml_helper.last_modification(url) -}}

**Pull request merged**